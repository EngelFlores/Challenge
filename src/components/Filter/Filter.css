.filter {
  cursor: pointer;
  position: relative;
  display: inline-block;
}

.filter-button{
  border: 0;
  width: 100%;
  background-color: white;
  height: 40px;
  font-size: 15px;
  border-radius: 20px;
}

.filter-list-container {
  display: none;
  position: absolute;
  background-color: #f9f9f9;
  min-width: 100px;
  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
  padding: 12px 16px;
  z-index: 1;
  margin-top: 5px;
  border-radius: 30px;
}

.show {
  display: block;
}
