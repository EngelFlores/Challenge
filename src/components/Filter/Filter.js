import React, { useState } from 'react';
import "./Filter.css"

const Filter = ({ title, handleFilter }) => {
  const [showMenu, setShowMenu] = useState(false);

  const handleMenu = () => {
    setShowMenu(!showMenu)
  }

  return (
    <div className="filter">
      <button className="filter-button" data-testid="filter-button" onClick={handleMenu}>{title}</button>
      <div className={`filter-list-container ${showMenu ? "show" : ""}`} data-testid="filter-list-container">
        <p className="filter-list-item" onClick={() => handleFilter("active")} data-testid="list-item-active">Active</p>
        <p className="filter-list-item" onClick={() => handleFilter("pending")} data-testid="list-item-pending">Pending</p>
        <p className="filter-list-item" onClick={() => handleFilter("declined")} data-testid="list-item-declined">Declined</p>
        <p className="filter-list-item" onClick={() => handleFilter("closed")} data-testid="list-item-closed">Closed</p>
      </div>
    </div>
  )
}

export { Filter }

