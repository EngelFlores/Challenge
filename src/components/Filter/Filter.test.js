import { render, waitFor, fireEvent } from '@testing-library/react';
import React from 'react'

import { Filter } from './Filter'

describe('Should render filter button', () => {
  test('Render filtered information from user', async () => {
    const { getByTestId } = render(<Filter title="Status" handleFilter={() => false} />)
    const filterButton = await waitFor(() => getByTestId('filter-button'))
    const listContainer = await waitFor(() => getByTestId('filter-list-container'))
    const listItemActive = await waitFor(() => getByTestId('list-item-active'))
    const listItemPending = await waitFor(() => getByTestId('list-item-pending'))
    const listItemDeclined = await waitFor(() => getByTestId('list-item-declined'))
    const listItemClosed = await waitFor(() => getByTestId('list-item-closed'))
    fireEvent.click(filterButton)

    expect(listContainer).toBeInTheDocument()
    expect(listItemActive).toBeInTheDocument()
    expect(listItemPending).toBeInTheDocument()
    expect(listItemDeclined).toBeInTheDocument()
    expect(listItemClosed).toBeInTheDocument()
  })

  test('Component match snapshot', async () => {
    const { getByTestId } = render(<Filter title="Status" handleFilter={() => false} />)
    const button = await waitFor(() => getByTestId('filter-button'))

    expect(button).toMatchSnapshot()
  })

})