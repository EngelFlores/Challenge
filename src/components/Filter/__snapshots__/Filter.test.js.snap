// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render filter button Component match snapshot 1`] = `
<button
  class="filter-button"
  data-testid="filter-button"
>
  Status
</button>
`;
